vortex-dtype/src/ptype.rs

```diff
@@ -598,38 +598,45 @@ macro_rules! match_each_native_simd_ptype {
 
 impl PType {
     /// Returns `true` iff this PType is an unsigned integer type
+    #[inline]
     pub const fn is_unsigned_int(self) -> bool {
         matches!(self, Self::U8 | Self::U16 | Self::U32 | Self::U64)
     }
 
     /// Returns `true` iff this PType is a signed integer type
+    #[inline]
     pub const fn is_signed_int(self) -> bool {
         matches!(self, Self::I8 | Self::I16 | Self::I32 | Self::I64)
     }
 
     /// Returns `true` iff this PType is an integer type
     /// Equivalent to `self.is_unsigned_int() || self.is_signed_int()`
+    #[inline]
     pub const fn is_int(self) -> bool {
         self.is_unsigned_int() || self.is_signed_int()
     }
 
     /// Returns `true` iff this PType is a floating point type
+    #[inline]
     pub const fn is_float(self) -> bool {
         matches!(self, Self::F16 | Self::F32 | Self::F64)
     }
 
     /// Returns the number of bytes in this PType
+    #[inline]
```

Comment:
I would do this for match_each macros?

Reply:
Keeping it for now, as we don't see any regressions, and the assumption is that this compiles to a lookup table.